docs/DeveloperGuide.adoc

```diff
@@ -377,6 +378,65 @@ Unable to differentiate between editing an existing value and adding a new value
 
 |===
 
+=== Storage of Medicine
+==== Current implementation
+
+The current implementation takes a similar form as the Windows file browser. The user is free to determine for himself/herself how he/she wants the medicines to be arranged.
+
+From the initial empty state of the storage, the users could arrange their storage in these following ways:
+
+1. The initial empty storage consists of an empty directory named as "root".
+
+2. The user could add a new directory via "addDirec" command by specifying the directory he/she wants to add into and the name of new directory.
+
+3. The current implementation does not allow multiple medicines with the same to exist simultaneously. However, one medicine could be placed in multiple directories.
+
+4. The user could add new/existing medicine to a specific directory via the "addMed" command.
+
+* 1. If there already exists a medicine with the same name in the storage, and the initial quantity is not specified in the command arguments, the existing medicine will be placed in the directory specified.
+
+* 2. Otherwise, a new medicine with the specified name, quantity and price will be created and added to the specified directory.
+
+5. The user could view detailed information about a directory or medicine through "viewMed" command.
+
+6. The user could set the alarm level for medicines via the "setAlarm", such that when the stored amount of the medicine falls lower than the alarm level, a reminder will be thrown.
+
+* 1. If the path in the command argument corresponds to a medicine, the medicine will have the alarm level specified.
+
+* 2. If the path in the command argument corresponds to a directory, every medicine under it and its sub-directories will have the alarm level specified. All medicines added under this directory will have the same alarm level.
+
+* 3. If the path neither points to a directory nor medicine, an error message will be displayed to the user.
+
+7. Additionally, a list of all medicine stored in the storage is also kept to enable fast search and manipulation of medicine by its name only. 
+
```

Comment:
If you have diagrams to support your implementation description, please create another folder in the docs/images directory so that we can better organize the images:

dg-[component]

When referencing the image in your description, just prepend the filepath of the folder where you store the image.